test/e2e/ocpbugs48050_test.go

```diff
@@ -0,0 +1,611 @@
+							VolumeMounts: []corev1.VolumeMount{
+								{
+									Name:      "cert",
+									MountPath: "/etc/serving-cert",
+								},
+							},
+						},
+					},
+					Volumes: []corev1.Volume{
+						{
+							Name: "cert",
+							VolumeSource: corev1.VolumeSource{
+								Secret: &corev1.SecretVolumeSource{
+									SecretName: "serving-cert-" + namespace,
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	if err := kclient.Create(context.TODO(), &deployment); err != nil {
+		return nil, err
+	}
+
+	return &deployment, nil
+}
+
+func createOCPBUGS48050Route(namespace, routeName, serviceName, targetPort string, terminationType routev1.TLSTerminationType) (*routev1.Route, error) {
+	route := routev1.Route{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      routeName,
+			Namespace: namespace,
+		},
+		Spec: routev1.RouteSpec{
+			To: routev1.RouteTargetReference{
+				Kind: "Service",
+				Name: serviceName,
+			},
+			Port: &routev1.RoutePort{
+				TargetPort: intstr.FromString(targetPort),
+			},
+			TLS: &routev1.TLSConfig{
+				Termination:                   terminationType,
+				InsecureEdgeTerminationPolicy: routev1.InsecureEdgeTerminationPolicyRedirect,
+			},
+			WildcardPolicy: routev1.WildcardPolicyNone,
+		},
+	}
+
+	if err := kclient.Create(context.TODO(), &route); err != nil {
+		return nil, fmt.Errorf("failed to create route %s/%s: %v", route.Namespace, route.Name, err)
+	}
+
+	return &route, nil
+}
+
+func setupOCPBUGS48050(t *testing.T, routeCount int) (*corev1.Namespace, error) {
+	var targetPorts = map[routev1.TLSTerminationType]string{
+		routev1.TLSTerminationEdge:        "http",
+		routev1.TLSTerminationReencrypt:   "https",
+		routev1.TLSTerminationPassthrough: "https",
+	}
+
+	var allTerminationTypes = []routev1.TLSTerminationType{
+		routev1.TLSTerminationEdge,
+		routev1.TLSTerminationReencrypt,
+		routev1.TLSTerminationPassthrough,
+	}
+
+	baseName := "ocpbugs40850-e2e"
+	ns := createNamespace(t, baseName+"-"+rand.String(5))
+
+	service, err := createOCPBUGS48050Service(ns.Name, baseName)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create service %s/%s: %v", ns.Name, baseName, err)
+	}
+
+	image, err := getCanaryImageFromIngressOperatorDeployment()
+	if err != nil || image == "" {
+		return nil, fmt.Errorf("failed to get canary image: %v", err)
+	}
+
+	deployment, err := createOCPBUGS48050Deployment(ns.Name, baseName, image)
+	if err != nil {
+		return nil, fmt.Errorf("failed to create deployment %s/%s: %v", ns.Name, baseName, err)
+	}
+
+	if err := waitForDeploymentComplete(t, kclient, deployment, 5*time.Minute); err != nil {
+		return nil, fmt.Errorf("deployment %s/%s not ready: %v", deployment.Namespace, deployment.Name, err)
+	}
+
+	makeRouteName := func(terminationType routev1.TLSTerminationType, i int) string {
+		return fmt.Sprintf("%s-route-%02d", string(terminationType), i)
+	}
+
+	for i := 0; i < routeCount; i++ {
+		for _, terminationType := range allTerminationTypes {
```

Comment:
Thanks for the observation! While the order isn't strictly important for functionality, it's easier to debug when the routes are created in a consistent order. Since maps in Go don't guarantee iteration order, I prefer using the `allTerminationTypes` slice to ensure predictable and repeatable behaviour during testing.

Reply:
Ok, closing this one.